packages/kiota_serialization_json/lib/src/json_serialization_writer.dart

```diff
@@ -0,0 +1,278 @@
+part of '../kiota_serialization_json.dart';
+
+class JsonSerializationWriter implements SerializationWriter {
+  final List<String> _buffer = [];
```

Comment:
There is a `StringBuffer` class that efficiently lets you build strings. Maybe we should use that instead https://api.dart.dev/stable/3.5.4/dart-core/StringBuffer-class.html

Reply:
@ricardoboss 

The StringBuffer class does not allow for deletions and we need to delete the separator character if it is the last one in the buffer (used in multiple places).
We could use Uint8Buffer as I have done for the MultiPart writer, resulting in code like this:
_buffer.addAll(utf8.encode('"${entry.key}":${entry.value}'));

Just let me know what you think.

Reply:
I left a new review